File: ios/Classes/Method.swift
//
//  Method.swift
//  in_app_purchase
//
//  Created by Bohdan Krokhmaliuk on 28.11.2021.
//

import Foundation

/// Incoming methods
enum Method: String {
    case initConnection = "init_connection"
    case endConnection = "end_connection"
    case setLogging = "enable_logging"
    case getInAppPurchases = "get_in_app_purchases"
    case getPurchasedProducts = "get_purchased_products"
    case startPurchase = "start_purchase"
    case finishTransaction = "finish_transaction"
    case requestReceipt = "request_receipt"
    case getAppStoreInitiatedProducts = "app_store_initiated_purchases"
    case getCachedInAppPurchases = "cached_in_app_purchases"
    case getPendingTransactions = "get_pending_transactions"
    case finishAllCompletedTransactions = "finish_completed_transactions"
}

/// Outgoing methods
enum OutMethod: String {
    case promotedProduct = "iap-promoted-product"
    case purchaseUpdate = "purchase-updated"
    case purchaseError = "purchase-error"
}

File: android/src/main/kotlin/com/kroha/in_app_purchase/Method.kt
package com.kroha.in_app_purchase

// Incoming methods
class Method {
    companion object {
        const val initConnection = "init_connection"
        const val endConnection = "end_connection"
        const val enableLogging = "enable_logging"
        const val startPurchase = "start_purchase"
        const val consumeProduct = "consume_product"
        const val getInAppPurchases = "get_in_app_purchases"
        const val acknowledgePurchase = "acknowledge_purchase"
        const val getPurchasedProductsByType = "get_purchased_products"
        const val consumeAllProducts = "consume_all_products"
        const val updateSubscription = "update_subscription"
        const val getPurchaseHistoryByType = "get_purchase_history"
    }
}

// Incoming methods
class OutMethod {
    companion object {
        const val purchaseError = "purchase-error"
        const val purchaseUpdate = "purchase-updated"
        const val connectionUpdate = "connection-updated"
    }
}